bokehjs/test/models/tools/edit/poly_draw_tool.ts

```diff
@@ -4,165 +4,221 @@ import * as sinon from "sinon"
 import {Keys} from "core/dom"
 import {create_1d_hit_test_result} from "core/hittest"
 
-import {Patches} from "models/glyphs/patches"
+import {Patches, PatchesView} from "models/glyphs/patches"
 import {Plot} from "models/plots/plot"
 import {Range1d} from "models/ranges/range1d"
 import {GlyphRenderer} from "models/renderers/glyph_renderer"
 import {ColumnDataSource} from "models/sources/column_data_source"
-import {PolyDrawTool} from "models/tools/edit/poly_draw_tool"
+import {PolyDrawTool, PolyDrawToolView} from "models/tools/edit/poly_draw_tool"
 
 const utils = require("../../../utils")
 
+export interface PolyDrawTestCase {
+  data: {[key: string]: (number[] | null)[]}
+  data_source: ColumnDataSource
+  draw_tool_view: PolyDrawToolView
+  glyph_view: PatchesView
+}
+
+export interface BkEv {
```

Comment:
I presume this could be just imported from appropriate tool's module?

Reply:
Yes, I actually wanted to ask you about that, these are currently duplicated all over the place. Is there a central place where we could define these or should I just eliminate the definitions in the tests for now?

Reply:
I did it on purpose to learn what all tools actually use (it's a mess with all sorts of behavior with hammer, native events and bokehjs' customizations). I've started typing ui_events and when done, event interfaces will be defined there. But for now just follow the existing approach.